proptools/tag.go

```diff
@@ -23,13 +23,20 @@ import (
 // HasTag returns true if a StructField has a tag in the form `name:"foo,value"`.
 func HasTag(field reflect.StructField, name, value string) bool {
 	tag := field.Tag.Get(name)
-	for _, entry := range strings.Split(tag, ",") {
-		if entry == value {
-			return true
-		}
-	}
+	return !strings.Contains(value, ",") && hasTagRecursive(tag, value)
+}
 
-	return false
+func hasTagRecursive(tag, value string) bool {
+	if strings.HasPrefix(tag, value) &&
+		(len(tag) == len(value) || len(tag) > len(value) && tag[len(value)] == ',') {
+		return true
+	} else if len(tag) <= len(value) {
+		return false
+	} else {
+		tag = tag[len(value):]
+		i := strings.Index(tag, ",")
+		return i != -1 && hasTagRecursive(tag[i+1:], value)
```

Comment:
Doesn't this need to call hasTagRecursive(tag, value) if `i == -1`?

Reply:
Also, would it be possible to write this iteratively?  It won't matter as along as it is using tail recursion, but it could easily get changed in the future to break the tail recursion and cause unexpectedly deep stack frames.